src/Microsoft.Health.Dicom.SqlServer/Features/ExtendedQueryTag/AddInstanceTableValuedParametersBuilder.cs

```diff
@@ -83,7 +83,7 @@ internal static class AddInstanceTableValuedParametersBuilder
 
         private static void AddPersonNameRow(DicomDataset instance, List<InsertPersonNameExtendedQueryTagTableTypeV1Row> personNamRows, QueryTag queryTag)
         {
-            string personNameVal = instance.GetSingleValueOrDefault<string>(queryTag.Tag);
+            string personNameVal = instance.GetSingleValueOrDefault<string>(queryTag.Tag, expectedVR: queryTag.VR);
```

Comment:
How could the VR be an unexpected value? Don't we only call these functions based on VR in `Build`? So isn't it already asserted that the VR type for the tag here is `PersonNameData` because we checked it beforehand?

Reply:
Discussed offline